File: app/src/main/java/com/alxad/sdk/demo/alx/RewardVideoActivity.java
package com.alxad.sdk.demo.alx;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.alxad.sdk.demo.AdConfig;
import com.alxad.sdk.demo.BaseActivity;
import com.alxad.sdk.demo.R;
import com.rixengine.api.AlxRewardVideoAD;
import com.rixengine.api.AlxRewardVideoADListener;

public class RewardVideoActivity extends BaseActivity implements View.OnClickListener {
    private static final String TAG = "AlxRewardVideoActivity";

    private TextView mTvTip;
    private TextView mTvShow;
    private AlxRewardVideoAD mVideoAD;
    private long startTime;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_load_and_show);
        setActionBar();
        initView();
    }

    private void initView() {
        TextView tv_video_load = findViewById(R.id.tv_load);
        mTvShow = findViewById(R.id.tv_show);
        mTvTip = findViewById(R.id.tv_tip);
        mTvShow.setEnabled(false);
        tv_video_load.setOnClickListener(this);
        mTvShow.setOnClickListener(this);
    }

    @SuppressLint("NonConstantResourceId")
    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.tv_load) {
            loadAd();
        }else if (v.getId() == R.id.tv_show) {
            if (mVideoAD == null) {
                Toast.makeText(this, getString(R.string.show_ad_no_load), Toast.LENGTH_SHORT).show();
                return;
            }
            if (mVideoAD.isReady()) {
                mVideoAD.showVideo(this);
            } else {
                loadAd();
            }
        }
    }

    /**
     * load Ad
     */
    public void loadAd() {
        mTvTip.setText(R.string.loading);
        startTime = System.currentTimeMillis();
        mVideoAD = new AlxRewardVideoAD();
        mVideoAD.load(this, AdConfig.ALX_REWARD_VIDEO_AD_ID, new AlxRewardVideoADListener() {

            @Override
            public void onRewardedVideoAdLoaded(AlxRewardVideoAD var1) {
                Log.i(TAG, "onRewardedVideoAdLoaded");
                mTvShow.setEnabled(true);
                Toast.makeText(getBaseContext(), getString(R.string.load_success), Toast.LENGTH_SHORT).show();
                mTvTip.setText(getString(R.string.format_load_success, (System.currentTimeMillis() - startTime) / 1000) + "｜ ecpm:" + mVideoAD.getPrice());

                mVideoAD.reportChargingUrl();
                mVideoAD.reportBiddingUrl();
            }

            @Override
            public void onRewardedVideoAdFailed(AlxRewardVideoAD var1, int errCode, String errMsg) {
                Log.i(TAG, "onRewardedVideoAdFailed：" + errCode + "; " + errMsg);
                mTvShow.setEnabled(false);
                Toast.makeText(getBaseContext(), getString(R.string.load_failed), Toast.LENGTH_SHORT).show();
                mTvTip.setText(R.string.load_failed);
            }

            @Override
            public void onRewardedVideoAdPlayStart(AlxRewardVideoAD var1) {
                Log.i(TAG, "onRewardedVideoAdPlayStart");
            }

            @Override
            public void onRewardedVideoAdPlayEnd(AlxRewardVideoAD var1) {
                Log.i(TAG, "onRewardedVideoAdPlayEnd");
            }

            @Override
            public void onRewardedVideoAdPlayFailed(AlxRewardVideoAD var2, int errCode, String errMsg) {
                Log.i(TAG, "onRewardedVideoAdPlayFailed:" + errCode + ";" + errMsg);
            }

            @Override
            public void onRewardedVideoAdClosed(AlxRewardVideoAD var1) {
                Log.i(TAG, "onRewardedVideoAdClosed");
                mTvShow.setEnabled(false);
                mTvTip.setText(R.string.tip_message);
            }

            @Override
            public void onRewardedVideoAdPlayClicked(AlxRewardVideoAD var1) {
                Log.i(TAG, "onRewardedVideoAdPlayClicked");
            }

            @Override
            public void onReward(AlxRewardVideoAD var1) {
                Log.i(TAG, "onReward");
            }

            @Override
            public void onRewardVideoCache(boolean isSuccess) {
                Log.i(TAG, "onRewardVideoCache:" + isSuccess + ";" + Thread.currentThread().getName());
            }
        });

    }

}

File: Demo App - Kotlin/app/src/main/java/com/alxad/sdk/demo/alx/RewardVideoActivity.kt
package com.alxad.sdk.demo.alx

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.TextView
import android.widget.Toast
import com.alxad.sdk.demo.AdConfig
import com.alxad.sdk.demo.BaseActivity
import com.alxad.sdk.demo.R
import com.rixengine.api.AlxAdParam
import com.rixengine.api.AlxRewardVideoAD
import com.rixengine.api.AlxRewardVideoADListener

class RewardVideoActivity : BaseActivity(), View.OnClickListener {

    override val TAG = "AlxRewardVideoActivity"

    private var mTvTip: TextView? = null
    private var mTvShow: TextView? = null
    private var mVideoAD: AlxRewardVideoAD? = null
    private var startTime: Long = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_load_and_show)
        setActionBar()
        initView()
    }

    private fun initView() {
        val tvLoad: TextView = findViewById(R.id.tv_load)
        mTvShow = findViewById(R.id.tv_show)
        mTvTip = findViewById(R.id.tv_tip)
        mTvShow?.isEnabled = false
        tvLoad.setOnClickListener(this)
        mTvShow?.setOnClickListener(this)
    }

    @SuppressLint("NonConstantResourceId")
    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_load -> loadAd()
            R.id.tv_show -> showAd()
        }
    }

    /**
     * load Ad
     */
    fun loadAd() {
        mTvTip?.setText(R.string.loading)
        startTime = System.currentTimeMillis()

        val userExtras: MutableMap<String, String> = HashMap()
        userExtras["bid_floor"] = "1.5"
        val builder = AlxAdParam.Builder().setUserExtras(userExtras)
        mVideoAD = AlxRewardVideoAD()
        mVideoAD?.load(
            this,
            AdConfig.ALX_REWARD_VIDEO_AD_ID,
            builder.build(),
            object : AlxRewardVideoADListener() {
                override fun onRewardedVideoAdLoaded(var1: AlxRewardVideoAD) {
                    Log.i(TAG, "onRewardedVideoAdLoaded")
                    Toast.makeText(
                        baseContext,
                        getString(R.string.load_success),
                        Toast.LENGTH_SHORT
                    ).show()
                    mTvShow?.isEnabled = true
                    mTvTip?.let {
                        val msg = getString(
                            R.string.format_load_success,
                            (System.currentTimeMillis() - startTime) / 1000
                        ) + "｜ ecpm:" + mVideoAD?.price
                        it.text = msg
                    }

                    mVideoAD?.reportChargingUrl()
                    mVideoAD?.reportBiddingUrl()
                }

                override fun onRewardedVideoAdFailed(
                    var1: AlxRewardVideoAD,
                    errCode: Int,
                    errMsg: String
                ) {
                    Log.i(TAG, "onRewardedVideoAdFailed：$errCode; $errMsg")
                    Toast.makeText(baseContext, getString(R.string.load_failed), Toast.LENGTH_SHORT)
                        .show()
                    mTvTip?.text = getString(R.string.load_failed)
                    mTvShow?.isEnabled = false
                }

                override fun onRewardedVideoAdPlayStart(var1: AlxRewardVideoAD) {
                    Log.i(TAG, "onRewardedVideoAdPlayStart")
                }

                override fun onRewardedVideoAdPlayEnd(var1: AlxRewardVideoAD) {
                    Log.i(TAG, "onRewardedVideoAdPlayEnd")
                }

                override fun onRewardedVideoAdPlayFailed(
                    var2: AlxRewardVideoAD,
                    errCode: Int,
                    errMsg: String
                ) {
                    Log.i(TAG, "onRewardedVideoAdPlayFailed:$errCode;$errMsg")
                }

                override fun onRewardedVideoAdClosed(var1: AlxRewardVideoAD) {
                    Log.i(TAG, "onRewardedVideoAdClosed")
                    mTvShow?.isEnabled = false
                    mTvTip?.setText(R.string.tip_message)
                }

                override fun onRewardedVideoAdPlayClicked(var1: AlxRewardVideoAD) {
                    Log.i(TAG, "onRewardedVideoAdPlayClicked")
                }

                override fun onReward(var1: AlxRewardVideoAD) {
                    Log.i(TAG, "onReward")
                }

                override fun onRewardVideoCache(isSuccess: Boolean) {
                    Log.i(
                        TAG,
                        "onRewardVideoCache:" + isSuccess + ";" + Thread.currentThread().name
                    )
                }
            })
    }

    private fun showAd(){
        if (mVideoAD == null) {
            Toast.makeText(this, getString(R.string.show_ad_no_load), Toast.LENGTH_SHORT).show()
            return
        }
        if (mVideoAD?.isReady == true) {
            mVideoAD?.showVideo(this)
        } else {
            loadAd()
        }
    }
}